mod/chain-spec/pkg/chain/chain_spec.go

```
@@ -485,8 +489,15 @@ func (c chainSpec[
 // withdrawals sweep.
 func (c chainSpec[
 	DomainTypeT, EpochT, ExecutionAddressT, SlotT, CometBFTConfigT,
-]) MaxValidatorsPerWithdrawalsSweep() uint64 {
-	return c.Data.MaxValidatorsPerWithdrawalsSweep
+]) MaxValidatorsPerWithdrawalsSweep(
+	isPostUpgrade func(uint64, SlotT) bool,
+	chainID uint64, slot SlotT,
+) uint64 {
+	if isPostUpgrade(chainID, slot) {
+		return c.Data.MaxValidatorsPerWithdrawalsSweepPostUpgrade
+	} else {
```

golangci-lint (1.62.2)

[warning] indent-error-flow: if block ends with a return statement, so drop this else and outdent its block

(revive)